client/session.go

```diff
@@ -106,6 +107,22 @@ type session struct {
 	streamBlocksBatchSize            int
 	streamBlocksMetadataBatchTimeout time.Duration
 	streamBlocksBatchTimeout         time.Duration
+	metrics                          sessionMetrics
+}
+
+type sessionMetrics struct {
+	sync.RWMutex
+	streamFromPeersMetrics map[uint32]*streamFromPeersMetrics
```

Comment:
Might be more memory efficient to use a value instead of a pointer for the metrics